jupyterhub_chameleon/utils.py

```diff
@@ -27,11 +27,13 @@ class Artifact:
             environment is torn down. Default = False.
     """
     def __init__(self, deposition_repo=None, deposition_id=None, id=None,
-                 contents_url=None, ownership='fork', ephemeral=None):
+                 contents_url=None, ownership='fork', ephemeral=None,
+                 contents_proto=None):
         self.id = id
         self.deposition_repo = deposition_repo
         self.deposition_id = deposition_id
         self.contents_url = contents_url
+        self.contents_proto = contents_proto
```

Comment:
This implies that proto cannot be derived from the URL? I think that's OK since git repos can be located at https:// urls, is that why it's like this?

Reply:
Yes, some git versions may have the HTTPS remote URL but the proto will be `git`.